### PR TITLE
Add tests & lint

### PR DESCRIPTION
**Pull request by adiprerepa**
[ X] Security

Adds tests to some security util functions.

### CODE REVIEW

File: pilot/pkg/security/trustdomain/util_test.go [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+			a:      "",
+			list:   []string{"yo", "tawg"},
+			expect: false,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.desc, func(t *testing.T) {
+			if got := stringMatch(tt.a, tt.list); got != tt.expect {
+				t.Errorf("%s: expect %v got %v", tt.desc, tt.expect, got)
```
Comment (nschhina) [inline pilot/pkg/security/trustdomain/util_test.go:63]: Nit: Can you change expect to expected

File: pilot/pkg/security/trustdomain/util_test.go [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+package trustdomain
+
+import (
+	"testing"
+)
+
+func TestStringMatch(t *testing.T) {
+	tests := []struct {
+		desc   string
+		a      string
```
Comment (nschhina) [inline pilot/pkg/security/trustdomain/util_test.go:24]: Nit: changing a to maybe something like "element" could be more descriptive

**Comment by adiprerepa** [comment]
@nschhina done, pls re-review.

**Comment by nschhina** [comment]
LGTM but let's wait on Security WG to approve this PR

**Pull request merged**